ui/app/AppLayouts/Browser/SignMessageModal.qml

```diff
@@ -0,0 +1,153 @@
+                    text: {
+                        switch(request.payload.method){
+                            case "personal_sign":
+                                return request.payload.params[0].length === bytes32Length ? request.payload.params[0] : utilsModel.hex2Ascii(request.payload.params[0]);
+                            case "eth_sign":
+                                return request.payload.params[1];
+                            case "eth_signTypedData":
+                            case "eth_signTypedData_v3":
+                                return JSON.stringify(request.payload.params[1]); // TODO: requires design
+                            default: 
+                                return JSON.stringify(request.payload.params); // support for any unhandled sign method 
+                        }
+                    }
+                }
+            }
+
+            TransactionSigner {
+                id: transactionSigner
+                width: stack.width
+                anchors.top: messageToSign.bottom
+                anchors.topMargin: Style.current.padding * 3
+                signingPhrase: walletModel.signingPhrase
+                reset: function() {
+                    signingPhrase = Qt.binding(function() { return walletModel.signingPhrase })
+                }
+            }
+        }
+    }
+
+    footer: Item {
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: parent.right
+        StyledButton {
+            id: btnBack
+            anchors.left: parent.left
+            width: 44
+            height: 44
+            visible: !stack.isFirstGroup
+            label: ""
+            background: Rectangle {
+                anchors.fill: parent
+                border.width: 0
+                radius: width / 2
+                color: btnBack.hovered ? Qt.darker(btnBack.btnColor, 1.1) : btnBack.btnColor
+
+                SVGImage {
+                    width: 20.42
+                    height: 15.75
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    anchors.verticalCenter: parent.verticalCenter
+                    fillMode: Image.PreserveAspectFit
+                    source: "../../img/arrow-right.svg"
+                    rotation: 180
+                }
+            }
+            onClicked: {
+                stack.back()
+            }
+        }
+        StatusButton {
+            id: btnNext
+            anchors.right: parent.right
+            //% "Next"
+            text: qsTrId("next")
+            enabled: stack.currentGroup.isValid && !stack.currentGroup.isPending
+            onClicked: {
+                const validity = stack.currentGroup.validate()
+                if (validity.isValid && !validity.isPending) {
+                    if (stack.isLastGroup) {
+                        return root.signMessage(transactionSigner.enteredPassword);
+                    }
+                    stack.next()
```

Comment:
You only have one group. You don't really need the stackView or calling next.

You can remove a lot of the components that use the stack IMO